inc/commondbtm.class.php

```diff
@@ -179,11 +194,15 @@ static function getTable($classname = null) {
          return '';
       }
 
-      if (empty($_SESSION['glpi_table_of'][$classname])) {
-         $_SESSION['glpi_table_of'][$classname] = getTableForItemType($classname);
+      $glpi_tables = self::getTablesOf();
+      if (!isset($glpi_tables[$classname]) || empty($glpi_tables[$classname])) {
+         $glpi_tables[$classname] = getTableForItemType($classname);
+         if ($GLPI_CACHE != null) {
+            $GLPI_CACHE->set('table_of', $glpi_tables);
```

Comment:
One thing here for this entire block, as it is repeated in other place, could it be better to place the `getTableForItemType` and the cache set in the `getTablesOf` function, no ?

Reply:
Where is this code repeated?

I do not understand what you have in mind, to me that's not really possible. Will talk about this wit you directly.